Add Computed Fields #4 (#577)
nautobot/nautobot: nautobot/docs/plugins/development.md

| `description` | Brief description of the plugin's purpose |
| `graphql_types` | The dotted path to the list of GraphQL type classes (default: `graphql.types.graphql_types)` |
| `installed_apps` | A list of additional Django application dependencies to automatically enable when the plugin is activated (you must still make sure these underlying dependent libraries are installed) |

> | `jinja_filters` | The path to the file that contains jinja filters to be registered (default: `jinja_filters`) |

Reviewer: ```suggestion
| `jinja_filters` | The dotted path to the submodule that contains jinja filters to be registered (default: `jinja_filters`) |
```

> | `jinja_filters` | The dotted path to the file that contains jinja filters to be registered (default: `jinja_filters.jinja_filters`) |

Reviewer: Remove? I don't see where any code is using this setting - it reads to me that jinja filters are being automatically discovered just by being included in a `jinja_filters.py`, with no need to explicitly list them in a separate `jinja_filters` list variable.

template_extensions = [SiteAnimalCount]
```

> ## Including Jinja Filters

Reviewer 2: ```suggestion
## Including Jinja2 Filters
```

> Plugins can define custom jinja filters to be used when rendering jinja in computed fields. Check out the [official Jinja documentation](https://jinja.palletsprojects.com/en/3.0.x/api/#custom-filters) on how to create filter functions.

Reviewer 2: ```suggestion
Plugins can define custom Jinja2 filters to be used when rendering templates defined in computed fields. Check out the [official Jinja2 documentation](https://jinja.palletsprojects.com/en/3.0.x/api/#custom-filters) on how to create filter functions.
```

In the file that defines your filters, you must import the `library` module from the `django_jinja` library. Filters must then be decorated with `@library.filter`. See an example below that defines a filter called `leet_speak`.

```python
from django_jinja import library

@library.filter
def leet_speak(input_str):
    charset = {"a": "4", "e": "3", "l": "1", "o": "0", "s": "5", "t": "7"}
    output_str = ""
    for char in input_str:
        output_str += charset.get(char.lower(), char)
    return output_str
```

> This filter will then be available for use in computed field templates like so:

Reviewer 2: ```suggestion
This filter will then be available for use in computed field templates like so:

```

```
{{ "HELLO WORLD" | leet_speak }}
```

> The output of this template results in the string `"H3110 W0R1D"`

Reviewer 2: ```suggestion
The output of this template results in the string `"H3110 W0R1D"`.
```
